mathesar_ui/src/stores/table-data/display.ts

```diff
@@ -30,23 +31,33 @@ const movementKeys = new Set([
   'Tab',
 ]);
 
+/**
+ * I don't understand what this is but I had to create it during a refactor.
+ * This value is used as a key in a ColumnPositionMap where column ids are also
+ * used, so should have the same type as a column id but needs to have a value
+ * that no column id will ever have. That's why we're using -1. If you
+ * understand this design and can suggest a better name for this variable, then
+ * please do so.
+ */
+export const ROW_POSITION_INDEX = -1;
+
```

Comment:
@pavish Would you like to lend a hand at clarifying this design? Maybe you could push a commit that cleans this up a bit by setting a better name for this constant and re-writing the comments above it? I also considered allowing strings as keys in the `ColumnPositionMap` so that we could keep the previous value of `__row`, but `-1` seemed a bit cleaner to me.

Reply:
Each entry within the columnPositionMap corresponds to the position of the column. However, the `_row` value consists of the total width & left values (i.e the position) of the row. It's placed within columnPositionMap inorder to avoid calculation within the component, which will run for each row.

We could use a dedicated store for it or even a new class containing both columnPosition and row width.

It was placed within `ColumnPositionMap` because we were passing around a lot of props to the child components and I wanted to reduce the number of props. Now, it is all passed down using context, so that's no longer an issue.

Reply:
Thanks for the explanation, @pavish. Are my changes in this PR with respect to `ColumnPositionMap` okay for now? Or would you like further adjustment? If you'd like to see more clean up here, can you either add a commit or let me know how you'd like me to address it?

Reply:
@seancolsen I think the change is fine. I wanted to specify it since you weren't sure about the context. You could just update your comment in this PR.